packages/core/src/notifications/controller.ts

```diff
@@ -0,0 +1,233 @@
+    private readonly state: NotificationsState
+    private readonly notificationsNode: NotificationsNode
+
+    static #instance: NotificationsController | undefined
+
+    constructor(extPrefix: 'amazonq' | 'toolkit', node: NotificationsNode) {
+        if (!NotificationsController.#instance) {
+            registerDismissCommand(extPrefix)
+        }
+        NotificationsController.#instance = this
+
+        this.storageKey = `aws.${extPrefix}.notifications`
+        this.notificationsNode = node
+
+        this.state = globals.globalState.get(this.storageKey) ?? {
+            startUp: {} as NotificationData,
+            emergency: {} as NotificationData,
+            dismissed: [],
+        }
+        this.state.startUp = this.state.startUp ?? {}
+        this.state.emergency = this.state.emergency ?? {}
+        this.state.dismissed = this.state.dismissed ?? []
+    }
+
+    public pollForStartUp(ruleEngine: RuleEngine) {
+        return this.poll(ruleEngine, 'startUp')
+    }
+
+    public pollForEmergencies(ruleEngine: RuleEngine) {
+        return this.poll(ruleEngine, 'emergency')
+    }
+
+    private async poll(ruleEngine: RuleEngine, category: NotificationType) {
+        try {
+            await this.fetchNotifications(category)
+        } catch (err: any) {
+            getLogger().error(`Unable to fetch %s notifications: %s`, category, err)
+        }
+
+        await this.displayNotifications(ruleEngine)
+    }
+
+    private async displayNotifications(ruleEngine: RuleEngine) {
+        const dismissed = new Set(this.state.dismissed)
+        const startUp =
+            this.state.startUp.payload?.notifications.filter(
+                (n) => !dismissed.has(n.id) && ruleEngine.shouldDisplayNotification(n)
+            ) ?? []
+        const emergency = (this.state.emergency.payload?.notifications ?? []).filter((n) =>
+            ruleEngine.shouldDisplayNotification(n)
+        )
+
+        NotificationsNode.instance.setNotifications(startUp, emergency)
+
+        // Emergency notifications can't be dismissed, but if the user minimizes the panel then
+        // we don't want to focus it each time we set the notification nodes.
+        // So we store it in dismissed once a focus has been fired for it.
+        const newEmergencies = emergency.map((n) => n.id).filter((id) => !dismissed.has(id))
+        if (newEmergencies.length > 0) {
+            this.state.dismissed = [...this.state.dismissed, ...newEmergencies]
+            await this.writeState()
+            void this.notificationsNode.focusPanel()
+        }
+    }
+
+    /**
+     * Permanently hides a notification from view. Only 'startUp' notifications can be dismissed.
+     * Users are able to collapse or hide the notifications panel in native VSC if they want to
+     * hide all notifications.
+     */
+    public async dismissNotification(notificationId: string) {
+        getLogger().debug('Dismissing notification: %s', notificationId)
```

Comment:
since logger accepts a topic now, it will help avoid some churn later to start using it where possible.
```suggestion
        getLogger('notifications').debug('Dismissing notification: %s', notificationId)
```

Reply:
If you want to add the topic its in `toolkitLogger.ts`